### PR TITLE
Add RENAME DATABASE

### PR DESCRIPTION
**Pull request by XisiHuang**
see #2034.
Still work in progress.
I wil need to support for `if exist` and return better error info when new name exists.
Tell me if you have other opinions.

### CODE REVIEW

File: sql/rename.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	if !dbDesc.HasPrivilege(p.user, parser.PrivilegeWrite) {
+		return nil, fmt.Errorf("user %s does not have %s privilege on database %s",
+			p.user, parser.PrivilegeWrite, dbDesc.Name)
+	}
+
+	descKey := structured.MakeDescMetadataKey(dbDesc.GetID())
```
Comment (MBerhault) [inline sql/rename.go]: the descriptor has the name as well, it needs to be updated.

File: sql/rename.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	"github.com/cockroachdb/cockroach/structured"
+)
+
+// RenameDatabase renames the database
+func (p *planner) RenameDatabase(n *parser.RenameDatabase) (planNode, error) {
+	if n.Name == "" {
+		return nil, errEmptyDatabaseName
+	}
+
+	dbDesc, err := p.getDatabaseDesc(string(n.Name))
```
Comment (MBerhault) [inline sql/rename.go]: let's make this require `root` for now, just like CreateDatabase.
Please add the following comment on the function:
```
privileges: "root" user.
   Notes: postgres requires superuser, db owner, or "CREATEDB".
          mysql >= 5.1.23 does not allow database renames.
```

File: sql/rename.go (hunk 1 above)
Comment (MBerhault) [inline sql/rename.go]: should add a test for that too. `SHOW DATABASES` just lists the names from the keys, but `SHOW GRANTS` for example uses DatabaseDescriptor.Name

File: sql/rename.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/sql/parser"
+	"github.com/cockroachdb/cockroach/structured"
+)
+
+// RenameDatabase renames the database
+func (p *planner) RenameDatabase(n *parser.RenameDatabase) (planNode, error) {
+	if n.Name == "" {
+		return nil, errEmptyDatabaseName
+	}
```
Comment (MBerhault) [inline sql/rename.go]: add a check for n.NewName == "" (shouldn't happen) and n.Name == n.NewName (noop)

**Comment by MBerhault** [comment]
you mention `if exists` in your description. There's no such option in postgres. I don't think we need it either. We do need a better error message when the new name is already taken though.

File: sql/testdata/rename [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+statement ok	
+SET DATABASE = u
+
+query II
+SELECT * FROM kv
+----
+1 2
+3 4
+5 6
+7 8
```
Comment (petermattis) [inline sql/testdata/rename]: Should add some error cases as well:

```
ALTER DATABASE "" RENAME to u
ALTER DATABASE u RENAME to ""
```

**Comment by petermattis** [comment]
LGTM

**Comment by tamird** [comment]
LGTM. We have previously discussed the notion of `ALTER DATABASE ALIAS a TO b` - it may be worth considering implementing `RENAME` in terms of `ALIAS` + `DROP`.

**Comment by XisiHuang** [comment]
PTAL, thanks

File: sql/rename.go [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+		//noop
+		return &valuesNode{}, nil
+	}
+
+	dbDesc, err := p.getDatabaseDesc(string(n.Name))
+	if err != nil {
+		return nil, err
+	}
+
+	if !dbDesc.HasPrivilege(p.user, parser.PrivilegeWrite) {
```
Comment (MBerhault) [inline sql/rename.go]: as mentioned before, this should check for "root" user, just like CreateDatabase.

File: sql/testdata/rename [hunk 6]
```diff
@@ -0,0 +1,74 @@
+query T
```
Comment (MBerhault) [inline sql/testdata/rename]: can you add a test either here, or in privilege_database to check that only root can rename a DB, even if the user has WRITE on the db

File: sql/rename.go (hunk 5 above)
Comment (XisiHuang) [inline sql/rename.go]: ah,miss it here.

File: sql/testdata/rename (hunk 6 above)
Comment (XisiHuang) [inline sql/testdata/rename]: ok, i'll add it. thanks your review.

**Comment by MBerhault** [comment]
LGTM

**Pull request merged**